src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -187,6 +191,7 @@ public String createSnapshot(
       // We fetch the DUOS dataset to confirm its existence, but do not need the returned value.
       duosClient.getDataset(duosId, userReq);
     }
+    validateForByRequestIdMode(snapshotRequestModel.getContents().get(0));
```

Comment:
I was thinking that this would be a general validation, although right now only `by request` mode is supported.
```suggestion
    validateSnapshotRequest(snapshotRequestModel.getContents().get(0));
```

My general thinking is that it's better to leave out details to make it easier to read the code. So here as a reader I'm thinking "this is doing some kind of validation, if I want to know more I can read the method". As a reader, if I see a more specific name, I'm thinking "If this is validating by request only, where are other validations being done?" All of this is subjective of course.

Reply:
So here is my thinking. Currently, this method first checks to ensure the mode is byRequestId and returns right away if it is not. It only does validation for the byRequestId mode. I think it might be beneficial to the reader for the method to have a more descriptive name that reflects what it currently does. If this is updated later to be a more broad validation step, then the name should be updated as well. Thoughts?

Reply:
It's hard to know what will be helpful for a future reader. My philosophy is to provide less detail whenever possible, because -- for me -- detail makes it harder to think in generalities, and -- also for me -- I find it easier to understand code when I can think generally about it.